core/domain/collection_domain.py

```diff
@@ -230,13 +263,54 @@ def create_default_node(cls, exploration_id):
         return cls(exploration_id)
 
 
+class CollectionDict(TypedDict, total=False):
+    """Type for the collection object"""
+
+    id: str
+    title: str
+    category: str
+    objective: str
+    language_code: str
+    tags: List[str]
+    schema_version: int
+    nodes: list[CollectionNodeDict]
+    version: int
+    created_on: Union[datetime.datetime, str, None]
+    last_updated: Union[datetime.datetime, str, None]
+
+
+class CollectionVersionDict(CollectionDict, total=False):
```

Comment:
Why `total=False`?

Reply:
we're changing the state of these keys by adding and deleting them according to different schema versions.
for eg:- here we are adding these keys.
  https://github.com/oppia/oppia/blob/add62e844665ab43229dbfd3cd207dc0e0343356/core/domain/collection_domain.py#L468

and here we are deleting these keys.
https://github.com/oppia/oppia/blob/add62e844665ab43229dbfd3cd207dc0e0343356/core/domain/collection_domain.py#L497

Reply:
total=false also removed from here !!

Reply:
I the `_convert_*` functions feel free to use ignores and only define the dict for the latest version that we use.

Reply:
can you please elaborate your comment!!, because i'm not able to get this part `I the _convert_* functions`.
may be i'm doing mistake ?

Reply:
Done!